.github/workflows/test.yml

```diff
@@ -40,22 +42,48 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: ./.github/actions/setup-android
-      - name: Start emulator
+      - name: Enable KVM
+        run: |
+          echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+          sudo udevadm control --reload-rules
+          sudo udevadm trigger --name-match=kvm
+
+      - name: Gradle cache
+        uses: gradle/actions/setup-gradle@v3
+
+      - name: AVD cache
+        uses: actions/cache@v3
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-31
```

Comment:
When using `actions/cache`, you need to use both `key` and `restore-keys`. Otherwise the cache will be saved once but never updated again. 

I recently used this action in iOS codebase. [Maybe this code could be a helpful reference for you?](https://github.com/customerio/customerio-ios/blob/490593e38d1bfa010b4361b9a4e1bc50c34a4230/.github/actions/main-sample-app-build/action.yml#L31-L62)